### PR TITLE
{ServiceConnector}: support function app in Service Connector passwordless

### PR DESCRIPTION
**Pull request by xfz11**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
ServiceConnector

File: src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py [hunk 1]
```diff
@@ -505,7 +506,7 @@ def create_aad_user(self):
                 if not self.ip:
                     error_code = ''
                     error_res = re.search(
-                        '\((\d{5})\)', str(e))
+                        r'\((\d{5})\)', str(e))
```
Comment (yungezz) [inline src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py:509]: why the change?

**Review by yungezz — APPROVED** [review]

File: src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py (hunk 1 above)
Comment (xfz11) [reply src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py:509]: lint error

**Comment by xfz11** [comment]
@kairu-ms Could you help merge the PR?

File: src/serviceconnector-passwordless/HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+0.3.14
```
Comment (jsntcy) [inline src/serviceconnector-passwordless/HISTORY.rst:5]: Now we support new versioning for extension and 0.x.x is not recommended any longer. The version will start from 1.0.0 (stable) or 1.0.0b1 (preview). You can find more details in https://github.com/Azure/azure-cli/blob/dev/doc/extensions/versioning_guidelines.md.

Comment (xfz11) [reply src/serviceconnector-passwordless/HISTORY.rst]: Updated. Thanks for reminding

**Review by jsntcy — APPROVED** [review]
